coordinator/core/core.go

```diff
@@ -101,56 +82,54 @@ func (c *Core) requireState(states ...state) error {
 	return errors.New("server is not in expected state")
 }
 
-func (c *Core) advanceState(newState state, tx store.Transaction) error {
-	txdata := storeWrapper{tx}
-	curState, err := txdata.getState()
+// AdvanceState advances the state of the Coordinator.
+func (c *Core) AdvanceState(newState state.State, tx store.Transaction) error {
+	txdata := wrapper.New(tx)
+	curState, err := txdata.GetState()
 	if err != nil {
 		return err
 	}
-	if !(curState < newState && newState < stateMax) {
+	if !(curState < newState && newState < state.Max) {
 		panic(fmt.Errorf("cannot advance from %d to %d", curState, newState))
 	}
-	return txdata.putState(newState)
+	return txdata.PutState(newState)
+}
+
+// Unlock the Core's mutex.
+func (c *Core) Unlock() {
```

Comment:
this is a bit shady, but we should keep it for now

Reply:
I added it to not alter existing code too much.
This is basically the old `c.mux.Unlock` that was used by the client API.
Ideally I would like to refactor the `RequireState` logic to not require us to defer an unlock statement beforehand.
Maybe this could be down by letting `RequireState` return an unlock function?

Reply:
> I added it to not alter existing code too much

Yeah, I thought so.

Let's keep it for now and maybe add a TODO and/or create a ticket to do it after we finished the more important tasks.